website/static/js/oauthAddonNodeConfig.js

```diff
@@ -0,0 +1,192 @@
+            return 'Successfully linked "' + $osf.htmlEscape(self.folder().name) + '". Go to the <a href="' +
+                self.urls().files + '">Files page</a> to view your content.';
+        });
+        var defaults = {
+            onPickFolder: function(evt, item) {
+                evt.preventDefault();
+                var name = item.data.path !== '/' ? item.data.path : '/ (Full ' + self.addonName + ')';
+                self.selected({
+                    name: name,
+                    path: item.data.path,
+                    id: item.data.id
+                });
+                return false; // Prevent event propagation
+            },
+            /**
+             * Allows a user to create an access token from the nodeSettings page
+             */
+            connectAccount: function() {
+                var self = this;
+
+                window.oauthComplete = function(res) {
+                    // Update view model based on response
+                    self.updateAccounts().done(function() {
+                        try{
+                            $osf.putJSON(
+                                self.urls().importAuth, {
+                                    external_account_id: self.accounts()[0].id
+                                }
+                            ).done(self.onImportSuccess.bind(self)
+                            ).fail(self.onImportError.bind(self));
+
+                            self.changeMessage(self.messages.connectAccountSuccess(), 'text-success', 3000);
+                        }
+                        catch(err){
+                            self.changeMessage(self.messages.connectAccountDenied(), 'text-failure', 6000);
+                        }
+                    });
+                };
+                window.open(self.urls().auth);
+            }
+        };
+        // Overrides
+        self.options = $.extend({}, defaults, opts);
+        // Treebeard config
+        self.treebeardOptions = $.extend(
+            {},
+            FolderPickerViewModel.prototype.treebeardOptions,
+            {
+                onPickFolder: function(evt, item) {
+                    return this.options.onPickFolder.call(this, evt, item);
+                }.bind(this),
+                resolveLazyloadUrl: function(item) {
+                    return item.data.urls.folders;
+                }
+            }
+        );
+    },
+    _updateCustomFields: function(settings){
+        this.validCredentials(settings.validCredentials);
+    },
+     /**
+     * Allows a user to create an access token from the nodeSettings page
+     */
+    connectAccount: function() {
+        this.options.connectAccount.call(this);
+    },
+    /**
+    * Imports addon settings from user's account. If multiple addon accounts are connected, allow user to pick between them.
+    */
+    importAuth: function(){
+        var self = this;
+
+        self.updateAccounts().done(function () {
```

Comment:
Does this work? It doesn't look like `updateAccounts` is returning anything. Please verify in an automated or manual test.